docs/usage.md

```diff
@@ -0,0 +1,277 @@
+3. workdir state (e.g. uncommitted changes since latest tag)
+
+and uses roughly the following logic to render the version:
+
+
+| distance | state     | format                                                               |
+|----------|-----------|----------------------------------------------------------------------|
+| no       | unchanged | `{tag}`                                                              |
+| yes      | unchanged | `{next_version}.dev{distance}+{scm letter}{revision hash}`           |
+| no       | changed   | `{tag}+dYYYYMMDD`                                                    |
+| yes      | changed   | `{next_version}.dev{distance}+{scm letter}{revision hash}.dYYYYMMDD` |
+
+where `{next_version}` is the next version number after the latest tag
+
+The next version is calculated by adding `1` to the last numeric component of
+the tag.
+
+For Git projects, the version relies on  [git describe](https://git-scm.com/docs/git-describe),
+so you will see an additional `g` prepended to the `{revision hash}`.
+
+
+!!! note
+
+    According to [PEP 440](https://peps.python.org/pep-0440/#local-version-identifiers>),
+    if a version includes a local component, the package cannot be published to public
+    package indexes like PyPI or TestPyPI. The disallowed version segments may
+    be seen in auto-publishing workflows or when a configuration mistake is made.
+
+    However, some package indexes such as devpi or other alternatives allow local
+    versions. Local version identifiers must comply with [PEP 440].
+
+## Semantic Versioning (SemVer)
+
+Due to the default behavior it's necessary to always include a
+patch version (the `3` in `1.2.3`), or else the automatic guessing
+will increment the wrong part of the SemVer (e.g. tag `2.0` results in
+`2.1.devX` instead of `2.0.1.devX`). So please make sure to tag
+accordingly.
+
+
+## Builtin mechanisms for obtaining version numbers
+
+1. the SCM itself (git/hg)
+2. `.hg_archival` files (mercurial archives)
+3. `.git_archival.txt` files (git archives, see subsection below)
+4. `PKG-INFO`
+
+
+### Git archives
+
+Git archives are supported, but a few changes to your repository are required.
+
+ensure the content of the following files:
+
+```{ .text file=".git_archival.txt"}
+
+node: $Format:%H$
+node-date: $Format:%cI$
+describe-name: $Format:%(describe:tags=true,match=*[0-9]*)$
+ref-names: $Format:%D$
+```
+
+``` {.text file=".gitattributes"}
+.git_archival.txt  export-subst
+```
+
+Finally, don't forget to commit those two files
+```commandline
+$ git add .git_archival.txt .gitattributes && git commit -m "add export config"
+```
+
+
+Note that if you are creating a `_version.py` file, note that it should not
+be kept in version control. It's strongly recommended to be put into gitignore.
+
+
+
+File finders hook makes most of MANIFEST.in unnecessary
+-------------------------------------------------------
```

Comment:
What level heading should this be? If not 3, put the right number of hashes here:
```suggestion
### File finders hook makes most of MANIFEST.in unnecessary
```